sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -199,7 +199,7 @@ case class ArraysZip(children: Seq[Expression]) extends Expression with ExpectsI
       """.stripMargin
     }
 
-    val splittedGetValuesAndCardinalities = ctx.splitExpressions(
+    val splittedGetValuesAndCardinalities = ctx.splitExpressionsWithCurrentInputs(
```

Comment:
LGTM.

Direct call of `splitExpressions` here has two problems:

1. `splitExpressions` can't work under wholestage codegen.
2. current row input is ignored in `arguments` for non wholestage codegen.

Reply:
yes, you're 100% right. That's why in the UT I added I am testing both cases.